cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java

```diff
@@ -380,9 +390,66 @@ public void testQueryMetrics() throws Exception {
         "&groupBy=namespace,flowlet&start=" + start + "&end=" + end, groupByResult);
   }
 
+
   @Test
-  public void testQueryMetricsWithTags() throws Exception {
+  public void testTimeRangeQueryBatch() throws Exception {
+    // note: times are in seconds, hence "divide by 1000";
+    long start = (emitTs - 60 * 1000) / 1000;
+    long end = (emitTs + 60 * 1000) / 1000;
 
+    verifyRangeQueryResult(
+      "/v3/metrics/query?" + getTags("yourspace", "WCount1", "WCounter", "splitter") +
+        "&metric=system.reads&start=" + start + "&end="
+        + end, 2, 3);
+
+    JsonObject q1 = getSingleQueryJson(ImmutableMap.of("namespace", "yourspace", "app", "WCount1", "flow", "WCounter",
+                                                       "flowlet", "splitter"),
+                                       ImmutableList.of("system.reads"), ImmutableList.<String>of(),
+                                       ImmutableMap.of("start", String.valueOf(start), "end", String.valueOf(end)));
+
+    JsonObject q2 = getSingleQueryJson(ImmutableMap.of("namespace", "yourspace", "app", "WCount1", "flow", "WCounter"),
+                                       ImmutableList.of("system.reads"), ImmutableList.<String>of("flowlet"),
+                                       ImmutableMap.of("start", String.valueOf(start), "end", String.valueOf(end)));
+    JsonArray queries = new JsonArray();
+    queries.add(q1);
+    queries.add(q2);
+    JsonObject taggedQuery = new JsonObject();
+    taggedQuery.add("timeRangeBatch", queries);
+    ImmutableMap<String, ImmutableList<ImmutableList<QueryResult>>> expected =
```

Comment:
we should have query result object that represents metrics query result - also in proto module